### PR TITLE
ci: add `clang-format` commit-hook

### PR DESCRIPTION
**Pull request by ysndr**
* add `clang-format` commit-hook
* ci: change git hooks to run on `push` rather than `commit`
  manually run them with 
    ```
    $ pre-commit run --hook-stage manual -a
    ``` 	
* chore: format c/c++ files

### CODE REVIEW

**Comment by ysndr** [comment]
some c++ things format kinda weirdly, but i dont quite know which knobs to dial in the `.clang-config`

File: pkgdb/.clang-format [hunk 1]
```diff
@@ -79,10 +79,10 @@ SpacesInSquareBrackets: false
 Standard: Auto
 TabWidth: 2
 UseTab: Never
-InsertBraces: true
+# InsertBraces: true
 EmptyLineBeforeAccessModifier: Always
 EmptyLineAfterAccessModifier: Always
-BreakBeforeConceptDeclarations: Always
+BreakBeforeConceptDeclarations: true
```
Comment (mkenigs) [inline pkgdb/.clang-format:85]: I don't really care about changes here, but I'll either let @aakropotkin  review them, or I can approve if you drop them

**Comment by mkenigs** [comment]
Thanks for moving checks to `push` rather than `commit`!

File: pkgdb/.clang-format (hunk 1 above)
Comment (ysndr) [reply pkgdb/.clang-format:85]: clang-format complained that `BreakBeforeConceptDeclarations` was not a boolean, so `true` it is i guess

File: flake.nix [hunk 2]
```diff
@@ -124,8 +124,16 @@
       # Generates a `.git/hooks/pre-commit' script.
       pre-commit-check = pre-commit-hooks.lib.${final.system}.run {
         src = builtins.path {path = ./.;};
+        default_stages = ["manual" "push"];
         hooks = {
           alejandra.enable = true;
+          clang-format = {
+            enable = true;
+            types_or = final.lib.mkForce [
+              "c"
```
[the remark this answers is not in the document]
Comment (ysndr) [reply flake.nix:133]: don't think so but i figured it's safer to als include it, i can leave it out if we want

File: pkgdb/.clang-format (hunk 1 above)
Comment (mkenigs) [reply pkgdb/.clang-format:85]: Hmm from the docs it seems `Always` should be okay? https://clang.llvm.org/docs/ClangFormatStyleOptions.html#breakbeforeconceptdeclarations

What about `InsertBraces`?

Comment (ysndr) [reply pkgdb/.clang-format:85]: didn't find it at all

We are also using clang-format `16.x.y` that documentation is about version `18`

Comment (mkenigs) [reply pkgdb/.clang-format:85]: Hmm I'm seeing it in 16.0.0? Is that still the wrong version?

https://releases.llvm.org/16.0.0/tools/clang/docs/ClangFormatStyleOptions.html#breakbeforeconceptdeclarations

https://releases.llvm.org/16.0.0/tools/clang/docs/ClangFormatStyleOptions.html#insertbraces

Comment (ysndr) [reply pkgdb/.clang-format:85]: correction: pre-commit-hook uses 14.0.6 (/nix/store/0r07ld5n1490rca3n2kldhca5ii5saz5-clang-tools-14.0.6/bin/clang-format) for some reason.
in that version [breakbeforeconceptdeclarations](https://releases.llvm.org/14.0.0/tools/clang/docs/ClangFormatStyleOptions.html#breakbeforeconceptdeclarations) was still a boolean 
image

and insertbraces does not exist.

let me see if i can update that

Comment (ysndr) [reply pkgdb/.clang-format]: ok we were explicitly using `clang-tools_16` in the shell.
I updated the pre-commit config to use that package as well and restored the .clang-format

**Review by mkenigs — APPROVED** [review]

**Pull request merged**